Microsoft.Toolkit.Uwp/Extensions/StringExtensions.cs

```diff
@@ -57,12 +57,22 @@ public static string GetLocalized(this string resourceKey, UIContext uiContext =
         }
 
         /// <summary>
-        /// Retrieves the provided resource for the given key for use independent of the UI thread.
+        /// Retrieves the provided resource for the given key for use independent of the UI thread. First looks up resource at the application level, before falling back to provided resourcePath. This allows for easily overridable resources within a library.
         /// </summary>
         /// <param name="resourceKey">Resource key to retrieve.</param>
-        /// <param name="resourcePath">Resource path to retrieve.</param>
+        /// <param name="resourcePath">Resource path to fall back to in case resourceKey not found in app resources.</param>
         /// <returns>string value for given resource or empty string if not found.</returns>
         public static string GetLocalized(this string resourceKey, string resourcePath)
-            => ResourceLoader.GetForViewIndependentUse(resourcePath).GetString(resourceKey);
+        {
+            // Try and retrieve resource at app level first.
+            var result = IndependentLoader.GetString(resourceKey);
```

Comment:
Hummmmm, have you tried this code path on Xaml Islands? We might need to use the other overload for GetLocalized, that accepts a UIContext, and force the use of this parameter, but I'm not 100% sure. Lets test it.

Reply:
@azchohfi I can just try this out over here, eh?

https://github.com/windows-toolkit/WindowsCommunityToolkit/blob/6a05a3c264407fbd1bee43fae57197a6e75af561/UnitTests/UnitTests.XamlIslands.UWPApp/XamlIslandsTest_StringExtensions.cs#L26-L42

I'll just add a test that doesn't provide the UIContext? I'll try copying the three tests I have over there...

Reply:
Yeah, create a test, just in case, and test it out on the Xaml Island test project, as the Xaml Islands tests are not automated into the CI.

Reply:
@azchohfi I tested this out and our API is working fine. However, the test itself was having trouble with the `ApplicationLanguages.ManifestLanguages.ToArray()` API as it only returned 1 result instead of 2 like in the default UWP cases... Any ideas? FYI @marb2000 

I'll check-in the updates I do have with that line commented out for now.